.azure-pipelines/testcases_collection/report_data_storage.py

```diff
@@ -0,0 +1,63 @@
+"""Wrappers and utilities for storing test cases."""
+import json
+import os
+import tempfile
+
+from azure.kusto.data import KustoConnectionStringBuilder
+from azure.kusto.ingest import QueuedIngestClient as KustoIngestClient
+from azure.kusto.ingest import IngestionProperties
+from azure.kusto.data.data_format import DataFormat
+from datetime import datetime
+
+
+class KustoConnector():
+    """KustoReportDB is a wrapper for storing test reports in Kusto/Azure Data Explorer."""
+
+    def __init__(self, db_name: str, db_table: str, db_table_mapping: str):
+        """Initialize a Kusto report DB connector.
+
+        Args:
+            db_name: The Kusto database to connect to.
+        """
+        self.db_name = db_name
+        self.db_table = db_table
+        self.db_table_mapping = db_table_mapping
+
+        ingest_cluster = os.getenv("TEST_REPORT_INGEST_KUSTO_CLUSTER_URL")
+        access_token = os.getenv('ACCESS_TOKEN', None)
+
+        if not ingest_cluster or not access_token:
+            raise RuntimeError(
+                "Could not load Kusto Credentials from environment")
+        else:
+            kcsb = KustoConnectionStringBuilder.with_aad_application_token_authentication(ingest_cluster, access_token)
+            self._ingestion_client = KustoIngestClient(kcsb)
+
+    def upload_testcases(self, test_cases):
```

Comment:
The uploading has been decoupled with "test case", there function name and argument do not have to be "test case" specific. They can be more generic, right?

Reply:
Changed to upload_results